posts/2020-05-15-five-years-of-rust.md

```diff
@@ -0,0 +1,497 @@
+contains APIs that don't require allocation or operating system level feature
+. The stabilisation of libcore brought the ability to compile Rust was one of
+the first major steps towards Rust's support for embedded systems development.
+
+**[1.10] — C ABI Dynamic Libraries** The `cdylib` crate type allows Rust to be
+compiled as a C dynamic library, enabling you to embed your Rust projects in
+any system that supports the C ABI. Some of Rust's biggest success stories
+among users is being able to write a small critical part of their system in
+Rust and seamlessly integrate in the larger codebase, and it's now easier
+than ever.
+
+**[1.12] — Cargo Workspaces** Workspaces allow you to organise multiple rust
+projects and share the same lockfile. Workspaces have been invaluable in
+building large multi-crate level projects.
+
+**[1.13] — The Try Operator** The first major syntax addition was the `?` or
+the "Try" operator. The operator allows you to easily propagate your error
+through your call stack. Previously you had to use the `try!` macro, which
+required you to wrap the entire expression each time you encountered a result,
+now you can easily chain methods with `?` instead.
+
+```rust
+try!(try!(expression).method()); // Old
+expression?.method()?;           // New
+```
+
+**[1.14] — Rustup 1.0** Rustup is Rust's Toolchain manager, it allows you to
+seamlessly use any version of Rust or any of its tooling. What started as a
+humble shell script has become what the maintainers affectionately call a
+*"chimera"*. Being able to provide first class compiler version management across
+Linux, macOS, Windows, and the dozens of target platforms would have been a
+myth just five years ago.
+
+#### 2017
+
+**[1.15] — Derive Procedural Macros** Derive Macros allow you to create powerful
+and extensive strongly typed APIs without all the boilerplate. This was the
+first version of Rust you could use libraries like `serde` or `diesel`'s
+derive macros on stable.
+
+**[1.17] — Rustbuild** One of the biggest improvements for our contributors to
+the language was moving our build system from the initial `make` based system
+to using cargo. This has opened up `rust-lang/rust` to being a lot easier for
+members and newcomers alike to build and contribute to the project.
+
+**[1.20] — Associated Constants** Previously constants could only be associated
+with a module. In 1.20 we stabilised associating constants on struct, enums,
+and importantly traits. Making it easier to add rich sets of preset values for
+types in your API, such as common IP addresses or interesting numbers.
+
+#### 2018
+
+**[1.24] — Incremental Compilation** Before 1.24 when you made a change in your
+library rustc would have to re-compile all of the code. Now rustc is a lot
+smarter about caching as much as possible and only needing to re-generate
+what's needed.
+
+**[1.26] — Existential Types** The addition of `impl Trait` gives you expressive
```

Comment:
This title should probably be `impl Trait` rather than existential types, since we never really called impl Trait that and because explicit existential types in associated types exists.

Reply:
Agreed.